pkg/pagerduty/pagerduty.go

```diff
@@ -126,6 +128,50 @@ func (p PagerDuty) AddNote(incidentID string, noteContent string) error {
 	return nil
 }
 
+// ExtractExternalIDFromCGHMAlertBody extracts from an Alert body until an external ID
+// the function is a member function but doesn't use any of the types
+func (_ PagerDuty) ExtractExternalIDFromCGHMAlertBody(data map[string]interface{}) (string, error) {
+
+	var err error
+	externalBody := externalCHGMAlertBody{}
+
+	err = mapstructure.Decode(data, &externalBody)
+	if err != nil {
+		return "", fmt.Errorf("cannot marshal externalCHGMAlertBody: %w", err)
+	}
+
+	internalRawData := externalBody.Details.Notes
```

Comment:
All in all, this looks ok to me. However, I am generally not sure about the definition of these structures for unmarshalling as there are just used in one place.
I would maybe just go the way over json/yaml unstructured here. But its probably just taste, so feel free to go ahead anyway: 

```
details, ok := data["details"].((map[string]interface{}))
if !ok{
    return nil, UnexpectedFormatErr{field="details", data=data}
}
notes := make(map[string]interface{})
err := yaml.Unmarshal([]byte(details), &notes)
if !err{
    return nil, UnexpectedFormatErr{field="Notes", data=details}
}
clusterid, ok :=  notes["ClusterId"]
if err != nil {
    return nil, UnexpectedFormatErr{field="ClusterId", data=notes}
}
return clusterid, nil

```

Reply:
I agree they are used only in one place, bit as @rafael-azevedo mentioned and I support that, marshalling to map[string]interface{} is bad

Reply:
why is it bad? Isn't a standard format to map a string key to different values:
 
`key : value with type of {string, int, float, map[string]interface{}}`

Reply:
not really. if I can get something more concrete here I will choose to do so. this is why I am willing to push the third parties in. having types creates certainty.

anyways we should move this offline